### PR TITLE
launch to production

### PR DESCRIPTION
**Pull request by Amrhub**
Changes featured in this PR @piero-vic  @Amrhub: 

- [x] navigation through the website using `<NavLInk />`
- [x] create all redux state slice files following `duck pattern`
- [x] upon first render fetching data from `rockets API`, only fetching once (navigating doesn't re-fetch) 
- [x] data stored about each rocket: `id, name, type, image, description, isReserved, and Wikipedia Link`
- [x] data stored about each mission: `id, name, description, reserved, and Wikipedia Link` (only fetched when you navigate to `Missions page`
- [x] rendering rocket page as per design also if there is no rocket fetch (fetch fails) it shows a message for user experience, also when loading
- [x] rendering missions page as per design
- [x] `Profile` page is used to show reserved rockets and booked missions using `filter( )`
- [x] updating badge and button according to `reserved` state in  rockets, missions, and profile pages
- [x] deployed netlify [dev branch](https://space-travelers-hub-pa.netlify.app/)

**NOTE**: The additional features are not integrated into the live demo, please check [additional features branch](https://github.com/Amrhub/space-travelers-hub/tree/additional-features) if you want to check it out
[Additional features Pull Request](https://github.com/Amrhub/space-travelers-hub/pull/68)

### CODE REVIEW

**Review by mke2111 — CHANGES_REQUESTED** [review]
### Hey @Amrhub & @piero-vic, 

**This is a great project you have going on, well-done lads**. 

   
**However, there are a few sections to improve before your project is approved.** 🔴 
<hr>

## Required Changes ♻️

- [ ] Unfortunately, you have not added the tests yet. Add unit tests with React Testing Library and Jest (for unit tests).
- [ ] Remove all the unnecessary files you are having in your app right now. (setuptests, reportwebvitals, and all the other files)

_Check the inline comments under the review for sections to improve._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Rockets/Rockets.jsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
+/* eslint-disable operator-linebreak */
+// import _ from 'lodash';
```
Comment (mke2111) [inline src/components/Rockets/Rockets.jsx]: - [ ] Remove this line of commented-out code if it is not being used by the component.

File: src/components/Missions/Missions.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+                </td>
+                <td className="mission-join">
+                  <Button
+                    isMissionJoined={mission.reserved}
+                    onClick={() => handleMissionReservation(mission.mission_id)}
+                  />
+                </td>
+              </tr>
+            );
+          })}
```
Comment (mke2111) [inline src/components/Missions/Missions.js]: - [ ] Just like in the rockets' component, use the react conditional rendering syntax while rendering the missions.
```
{mission.reserved && ( 
    // qwerty
)}

```

**Comment by Amrhub** [comment]
Hi @mke2111, 
I agree with most of the changes required, but, tests are not part of the requirements 🤔, Could you please confirm?

**Comment by mke2111** [comment]
Oh yes, @Amrhub, you must have missed this requirement.
https://github.com/microverseinc/curriculum-react-redux/blob/main/group-project/project_space_travelers_hub.md
Screenshot from 2021-12-17 21-25-06

Screenshot from 2021-12-17 21-27-55

**Comment by Amrhub** [comment]
I think so, but, because we haven't had any cards ensuring this or time-space for that and my other colleagues got approved without testing. 
image
according to this line, `All tasks should be based on the cards from your Kanban board.` I didn't implement the testing, neither did my other colleagues. 
If we have it as part of requirements it always has space and taken into consideration in the kanban

**Review by D4nielJ — APPROVED** [review]
## Hello @Amrhub and @piero-vic, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉



## Optional
- [ ] I'm sorry there are discrepancies in the curriculum. In this case, I have to agree with you and I will leave the tests optional. However, I do recommend testing the app, this is an ability that is highly appreciated by recruiters and will probably be part of your daily tasks working on any project.

On my side, I will raise a ticket, and hopefully, future students won't be confused about the requirements.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can always find me on slack!

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Amrhub** [comment]
Totally agree, I don't mind working on tests but when you are left with no time for it then it's hard to implement it last minute 🙂

**Pull request merged**